wicket-core/src/main/java/org/apache/wicket/markup/html/form/upload/FileUploadField.java

```diff
@@ -47,6 +57,74 @@
 {
 	private static final long serialVersionUID = 1L;
 
+	/**
+	 *  {@link org.apache.wicket.ajax.form.OnChangeAjaxBehavior} that streams back to server properties
+	 *  of the selected file (at client side), even when file has not yet being uploaded.
+s	 *
+	 * @author Ernesto Reinaldo Barreiro (reiern70@gmail.com).
+	 */
+	public static abstract class OnFileSelectedBehavior extends OnChangeAjaxBehavior
+	{
+		private static final long serialVersionUID = 1L;
+
+		@Override
+		protected void onBind() {
+			super.onBind();
+			Component component = getComponent();
+			if (!(component instanceof FileUploadField))
+			{
+				throw new WicketRuntimeException("Behavior " + getClass().getName()
+						+ " can only be added to an instance of a FileUploadField");
+			}
+		}
+
+		@Override
+		protected void onUpdate(AjaxRequestTarget target)
+		{
+			Request request = RequestCycle.get().getRequest();
+			IRequestParameters parameters = request.getRequestParameters();
+			String fileName = parameters.getParameterValue("fileName").toString();
+			long fileSize = parameters.getParameterValue("fileSize").toLong(0);
+			Date lastModified = new Date(parameters.getParameterValue("lastModified").toLong(0));
+			String mimeType = parameters.getParameterValue("mimeType").toString();
+			onFileSelected(target, fileName, fileSize, lastModified, mimeType);
+		}
+
+		/**
+		 * Called when a file, at client side is selected.
+		 *
+		 * @param target The {@link org.apache.wicket.ajax.AjaxRequestTarget}
+		 * @param fileName The client file name
+		 * @param fileSize The client file size
+		 * @param lastModified The {@link java.util.Date} when file was last modified.
+		 * @param mimeType The  MIME type of file.
+		 */
+		protected abstract void onFileSelected(AjaxRequestTarget target, String fileName, long fileSize,
```

Comment:
I think we should think about the `multiple` support now, because later this method signature may need to be changed to something like `protected abstract void onFilesSelected(AjaxRequestTarget target, List<FileDescription>)`

Reply:
Oh. I was thinking to roll a different behavior. But you are probably right. I will add support for multiple values ASAP

Reply:
My latest commit adds support for multiple files. I have tested it locally and it seems to work (with multiple files). Not sure if the way I did it is the best/correct one.